src/app/pages/storage/disks/disk-form/disk-form.component.ts

```diff
@@ -1,220 +1,133 @@
-        {
-          type: 'checkbox',
-          name: 'clear_pw',
-          placeholder: helptext.clear_pw.placeholder,
-          tooltip: helptext.clear_pw.tooltip,
-        },
-      ],
-    },
-    { name: 'divider', divider: true },
-  ];
-
-  title: string;
-
-  rowid: string;
+export class DiskFormComponent implements OnInit {
+  form = this.fb.group({
+    name: [''],
+    serial: [''],
+    description: [''],
+    critical: [null as number, [Validators.min(0)]],
+    difference: [null as number, [Validators.min(0)]],
+    informational: [null as number, [Validators.min(0)]],
+    hddstandby: [null as DiskStandby],
+    advpowermgmt: [null as DiskPowerLevel],
+    togglesmart: [false],
+    smartoptions: [],
+    passwd: [''],
+    clear_pw: [false],
+  });
+  readonly helptext = helptext;
+  readonly title = helptext.disk_form_title;
+  readonly hddstandbyOptions$ = of(helptext.disk_form_hddstandby_options);
+  readonly advpowermgmtOptions$ = of(this.translateOptions(this.helptext.disk_form_advpowermgmt_options));
+  isLoading = false;
+  existingDisk: Disk;
 
   constructor(
-    private router: Router,
-    protected ws: WebSocketService,
-    protected aroute: ActivatedRoute,
+    private translate: TranslateService,
+    private ws: WebSocketService,
+    private fb: FormBuilder,
+    private dialogService: DialogService,
+    private cdr: ChangeDetectorRef,
+    private errorHandler: FormErrorHandlerService,
+    private slideInService: IxSlideInService,
   ) {
   }
 
-  resourceTransformIncomingRestData(data: Disk): Disk {
-    const transformed = { ...data };
-    delete transformed.passwd;
-    return transformed;
+  ngOnInit(): void {
+    this.clearPasswordField();
+  }
+
+  setFormDisk(disk: Disk): void {
+    this.existingDisk = disk;
+    const transformedData = { ...disk };
+    delete transformedData.passwd;
+    this.form.patchValue(transformedData);
   }
 
-  preInit(): void {
-    this.aroute.params.pipe(untilDestroyed(this)).subscribe((params) => {
-      /*
-       * Make sure the route is "storage/disks" before
-       * using the pk value
-       * */
-      if (params['pk'] && this.router.url.startsWith('/storage/disks')) {
-        this.customFilter[0][0].push(params['pk']);
-      }
+  private translateOptions(options: Option[]): Option[] {
+    return options.map((option) => {
+      return { label: this.translate.instant(option.label), value: option.value };
     });
   }
 
-  beforeSubmit(value: any): void {
-    if (value.passwd === '') {
-      delete value.passwd;
+  private clearPasswordField(): void {
+    this.form.controls['clear_pw'].valueChanges
+      .pipe(untilDestroyed(this))
+      .subscribe(
+        (state) => {
+          const controlPasswd = this.form.controls['passwd'];
+          if (state) {
+            controlPasswd.disable();
+          } else {
+            controlPasswd.enable();
+          }
+        },
+      );
+  }
+
+  transformFormValue(value: any): DiskUpdate {
```

Comment:
We try to avoid using `any` as much as possible. In this case it can be typed to `DiskFormComponent['form']['value']`